### ISSUE TITLE
DAUTO_SAVE_PAGING_COUNT is not effective anymore

### ISSUE THREAD

**Issue by MondoGao** [koreader/koreader#11504]
* KOReader version: 2024.01-56
* Device: Huawei MatePad Paper 10'

#### Issue

DAUTO_SAVE_PAGING_COUNT is not effective anymore.

The latest modification to this setting is https://github.com/koreader/koreader/commit/014abf21bc50c22bfebf583ccb1be16bd8533043#diff-d5c59944a61752979b90d222e178b78cf71e4b29d8a6714a6c7a6d6decc21204

but the same code is now on master now

https://github.com/koreader/koreader/blob/master/frontend/apps/reader/modules/readerview.lua#L1147

#### Steps to reproduce
```lua
-- default.persistent.lua
return {
DAUTO_SAVE_PAGING_COUNT = 0,
}
```

Try any pdf annotation and page up and down.

**Comment by MondoGao** [comment]
Find this pr remove this setting, https://github.com/koreader/koreader/pull/5687.
But how can I save the pdf annotation instantly now without this setting?

https://github.com/koreader/koreader/issues/2258

**Comment by Frenzie** [comment]
It was deprecated five years ago in favor of a UI setting in <https://github.com/koreader/koreader/pull/5687>. It remained commented out with a note saying `no longer needed (now available in menu as an interval in minutes)` for a few years until it was removed two years ago in https://github.com/koreader/koreader/pull/9546

**Comment by MondoGao** [comment]
> It was deprecated five years ago in favor of a UI setting in #5687. It remained commented out with a note saying `no longer needed (now available in menu as an interval in minutes)` for a few years until it was removed two years ago in #9546

Yeah I just found that out. But in that case how can I achieve same effect with  DAUTO_SAVE_PAGING_COUNT = 0? Like discussion in https://github.com/koreader/koreader/issues/2258

**Comment by Frenzie** [comment]
I think something like this isn't half bad. It'll immediately save when changing the page, but not after every highlight on the same page.

```lua
-- 2-save-next-page-change-after-highlight.lua
local Event = require("ui/event")
local UIManager = require("ui/uimanager")
local time = require("ui/time")

local ReaderHighlight = require("apps/reader/modules/readerhighlight")
local ReaderView = require("apps/reader/modules/readerview")

local save_new_page_update = false

local saveHighlight_orig = ReaderHighlight.saveHighlight

ReaderHighlight.saveHighlight = function(self, action, page, item, content)
    -- Run original code
    saveHighlight_orig(self, action, page, item, content)

    save_next_page_update = true
end

local onPageUpdate_orig = ReaderView.onPageUpdate

ReaderView.onPageUpdate = function(self, new_page_no)
    -- Run original code
    onPageUpdate_orig(self, new_page_no)

    if save_next_page_update then
        if not self.settings_last_save_time then -- reader not yet ready
            return
        end
        if G_reader_settings:nilOrFalse("auto_save_settings_interval_minutes") then
            -- no auto save
            return
        end

        local interval_m = G_reader_settings:readSetting("auto_save_settings_interval_minutes")
        local interval = time.s(interval_m * 60)
        local now = UIManager:getElapsedTimeSinceBoot()
        self.settings_last_save_time = now - interval
        save_next_page_update = false
    end
end
```

**Comment by MondoGao** [comment]
I'll test that out later tonight, thanks a lot!

**Comment by ER-95** [comment]
@Frenzie Sorry to bump an old topic, but are there any pros/cons of using your autosave method vs editing the settings.reader.lua file to autosave every minute instead of every 5 minutes? I'm having this same issue but unsure which route to take.

Many thanks

**Comment by Frenzie** [comment]
Using hacks is only recommended if you like playing with hacks. ;-) But the code I wrote above should be reasonably future-proof because it runs some extra code after executing the original function, rather than overriding it completely.
